lib/apiservers/engine/proxy/archive.go

```diff
@@ -123,7 +122,7 @@ func (a *ArchiveProxy) ArchiveExportReader(ctx context.Context, store, ancestorS
 
 // ArchiveImportWriter initializes a write stream for a path.  This is usually called
 // for getting a writer during docker cp TO container.
-func (a *ArchiveProxy) ArchiveImportWriter(ctx context.Context, store, deviceID string, filterSpec vicarchive.FilterSpec) (io.WriteCloser, error) {
+func (a *ArchiveProxy) ArchiveImportWriter(op trace.Operation, store, deviceID string, filterSpec vicarchive.FilterSpec, errchan chan error) (io.WriteCloser, error) {
 	defer trace.End(trace.Begin(deviceID))
 
 	if store == "" || deviceID == "" {
```

Comment:
If this ends up being the case we would return before populating the `errchan` which could cause us to wait forever later on.